### PR TITLE
Libressl update to 3.2.2 (stable)/ Fix regression

### PR DESCRIPTION
**Pull request by satmandu**
Sync with stable version tracking OpenBSD 6.8
Works properly:
- [x] x86_64

### CODE REVIEW

**Comment by uberhacker** [comment]
Didn't we just update libressl?  You do realize this update will break everything.  We'll likely need to recompile python, php (4 versions) and a slew of other packages to get this to work.

**Comment by satmandu** [comment]
I think there was a discussion of keeping some of the security apps up to date? libressl would definitely be one of those.

This doesn't appear to break anything though, as the same library names are used. (But feel free to prove me wrong.)

**Comment by satmandu** [comment]
@uberhacker Do we need more complicated logic in making symlinks to support older installs?

libressl.so.#.0.1 => libressl.so.1.0.0?

File: packages/libressl.rb [hunk 1]
```diff
@@ -24,9 +24,15 @@ def self.install
...
+
+    # add symlink to libcrypto.so.1.0.0
+    #FileUtils.ln_s "#{CREW_LIB_PREFIX}/libcrypto.so.1.0.0", "#{CREW_DEST_LIB_PREFIX}/libcrypto.so.46"
+
+    # add symlink to libssl.so.1.0.0
+    FileUtils.ln_s "#{CREW_LIB_PREFIX}/libssl.so.1.0.0", "#{CREW_DEST_LIB_PREFIX}/libssl.so.47"
 
     # add symlink to libssl.so.1.0.0
-    FileUtils.ln_s "#{CREW_LIB_PREFIX}/libssl.so.48.0.1", "#{CREW_DEST_LIB_PREFIX}/libssl.so.1.0.0"
+    #FileUtils.ln_s "#{CREW_LIB_PREFIX}/libssl.so.1.0.0", "#{CREW_DEST_LIB_PREFIX}/libssl.so.48"
```
Comment (uberhacker) [inline packages/libressl.rb:36]: It doesn't hurt to have all these symlinks but a better solution would be to recompile all packages that rely on the older version to link to the newer version.

**Comment by uberhacker** [comment]
@satmandu:  My energy level to test this update is pretty low.  Last time we updated this package, we needed to recompile or update numerous other packages.  It opens a Pandora's box.  This is a core package so we will need binaries for the install along with any other core packages this update affects.

**Comment by satmandu** [comment]
@uberhacker Are you doing all of these compiles on native hardware for each of these systems?


I still think last month's libressl update was fundamentally broken because we didn't create symlinks properly.

 ```   FileUtils.ln_s "#{CREW_LIB_PREFIX}/libssl.so.48.0.1", "#{CREW_DEST_LIB_PREFIX}/libssl.so.1.0.0"``` is wrong.
That symlink is _automatically created anyways_. 

**What we need is** 
 ```   FileUtils.ln_s "#{CREW_LIB_PREFIX}/libssl.so.1.0.0", "#{CREW_DEST_LIB_PREFIX}/libssl.so.47"``` so that older installs would keep working. Instead the update borked any program looking for ```#{CREW_DEST_LIB_PREFIX}/libssl.so.47```.

If you don't want to put this in now, you might want to rebuild the libressl 3.1.4 package anyways so that symlink is there, for anyone who upgrades in the meantime.

At least, that's my thinking.

**Comment by uberhacker** [comment]
A more effective solution would be to recompile the affected packages.  The main question:  How many other packages are affected?  It may be in the dozens.  The good thing is most of those packages probably need updated to newer versions anyway.

**Comment by vincowl** [comment]
Just a quick intrusion : when we updated `qtbase` we had to disable openssl support since qt doesn't support libressl. Recent libressl update didn't solve the problem. Keeping stuck with libressl will imply no ssl support in the qt packages.... If everything should be recompiled, won't it be a good opportunity to roll back to openssl ? Do we have benefits that I don't get (other than philosophical ones) to stay with libressl ?

**Comment by satmandu** [comment]
> A more effective solution would be to recompile the affected packages. The main question: How many other packages are affected? It may be in the dozens. The good thing is most of those packages probably need updated to newer versions anyway.

Best way would be to spin up a container and just recursively build all dependencies.

If you use my script and go to /output/ and then do a recursive build of all downstream dependencies, all the resultant files will drop in the folder you invoked the docker container from.

Ideally one would have a list of symbols  that apps and libraries in a package both use and provide, and one could easily figure out what the dependencies are. Debian does that well...

**Comment by uberhacker** [comment]
We definitely need a better dependency resolver.  That is [one of the Hacktoberfest challenges](https://github.com/skycocker/chromebrew/issues/4338).  You would be a hero if you can figure it out.

**Comment by satmandu** [comment]
I feel like this is a problem that has been solved by multiple distributions already. Why reinvent the wheel?

I'm in the automate or reuse code as much as possible to avoid busywork camp.

(This is also why I was happy to spend time to setup docker containers so I wouldn't have to do the same work over and over again.)

**Comment by uberhacker** [comment]
OK, we won't reinvent the wheel.  Now figure out a way to leverage how other package managers handle dependencies and incorporate in this project. :)

**Comment by satmandu** [comment]
See https://github.com/Homebrew/brew/blob/master/Library/Homebrew/dependency_collector.rb
but also https://blog.jpalardy.com/posts/untangling-your-homebrew-dependencies/

Isn't chromebrew spiritually a descendant of the homebrew project? It's also in ruby, so it seems it would be _easiest_ [not actually easy] to leverage their infrastructure? At least their setup is also rpedicated upon installing alongside an existing system instead of using a chroot.

**Comment by uberhacker** [comment]
Addressed in #4532.

**Pull request closed without merging**